clients/bazel-module-registry/src/main/kotlin/BazelModuleRegistryService.kt

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright (C) 2020 The ORT Project Authors (see <https://github.com/oss-review-toolkit/ort/blob/main/NOTICE>)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.clients.bazelmoduleregistry
+
+import com.jakewharton.retrofit2.converter.kotlinx.serialization.asConverterFactory
+
+import java.net.URI
+import java.util.Base64
+
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.Serializer
+import kotlinx.serialization.encoding.Decoder
+import kotlinx.serialization.encoding.Encoder
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.JsonNamingStrategy
+
+import okhttp3.MediaType.Companion.toMediaType
+import okhttp3.OkHttpClient
+
+import retrofit2.Retrofit
+import retrofit2.http.GET
+import retrofit2.http.Path
+
+import org.ossreviewtoolkit.model.Hash
+import org.ossreviewtoolkit.model.HashAlgorithm
+import org.ossreviewtoolkit.model.RemoteArtifact
+import org.ossreviewtoolkit.model.VcsInfo
+import org.ossreviewtoolkit.model.VcsType
+
+@Serializer(URI::class)
+object URISerializer : KSerializer<URI> {
+    override fun serialize(encoder: Encoder, value: URI) = encoder.encodeString(value.toString())
+    override fun deserialize(decoder: Decoder) = URI(decoder.decodeString())
+}
+
+/**
```

Comment:
I propose to do the following with the "model" classes.

1. Move all model classes to the top / file level. Optionally into a separate dedicated file.
2. Rename ModuleMaintainer to just Maintainer and move it inside ModuleMetadata
3. Rename ModuleSource to ModuleSourceInfo to align with the naming of the API